### ISSUE TITLE
"Jumping" to a value then springing back to original value quickly does not animate

### ISSUE THREAD

**Issue by BrennanRoberts** [chenglou/react-motion#179]
Let's say you have a render function like this:
```javascript
Motion({
  style: {
    opacity: this.props.visible ? 1 : spring(0)
  },
} ...);
```
The intent is to have the component show instantaneously, but fade out.

If this component is passed `visible: false, true, false` quickly, then no animation will ever occur.

JSFiddle here:
http://jsfiddle.net/17hdch0w/

**Comment by chenglou** [comment]
Yeah that's the desired behaviour. The spring's initial value is the current value. When you quickly switch like that it attempts to go to 1 (e.g. 0.001), then quickly goes back to 0.

We don't allow overriding current value _yet_. If this is for the first render, use `defaultStyle`.

**Comment by BrennanRoberts** [comment]
Thanks for the quick response, @chenglou. Forgive me if I'm wrong, but based on your response, I think there's a chance you've misunderstood. In the example, we're attempting to skip straight to `1`; thus the omission of `spring()`.

If this truly is the desired behavior, would you agree that the docs are misleading when they say: 
"If a plain number is provided rather than a spring config, instead of giving an interpolated value in the children function param below, we'll jump straight to that number value."

Thanks for your work on this library!

**Comment by chenglou** [comment]
Ah, that's a bug then. @bsansouci

**Comment by chenglou** [comment]
Just a heads up, I've fixed this locally (finally...) but it was a chore. Combining the logic of spring and non-spring values for the same prop was tedious.

It works now, but I'm thinking more and more that the use-case doesn't justify it. Can you give me the concrete thing you're trying to do here? Maybe there's a better way of handling it.

**Comment by BrennanRoberts** [comment]
Sure. I've got a piece of modal "pending" UI that only lasts for a split-second, but that I'd like to be visible long enough to be read. Its `opacity` gets `spring(0)` when inactive and `1` when active, so that it jumps in and fades out.

I could do a near-instantaneous fade in using a spring in this case, but figured this would eventually bite others, as it contradicts the docs.

**Comment by chenglou** [comment]
Fixed in v0.4.1. Sorry for the wait!
